libs/components/indicators/src/lib/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -9,7 +9,94 @@ import {
 } from '@angular/core';
 import { MutationObserverService } from '@skyux/core';
 
-const className = 'sky-highlight-mark';
+const CLASS_NAME = 'sky-highlight-mark';
+
+function markNode(node: Text, searchRegex: RegExp): number {
+  // The search regular expression is reused across calls to markNode(), so reset
+  // it so it searches from the start of the string each time.
+  searchRegex.lastIndex = 0;
+
+  const text = node.nodeValue;
+
+  if (text) {
+    const match = searchRegex.exec(text);
+
+    if (match) {
+      // Split apart text node with mark tags in the middle on the search term.
+      const matchIndex = match.index;
+
+      const middle = node.splitText(matchIndex);
+      middle.splitText(searchRegex.lastIndex - matchIndex);
+      const middleClone = middle.cloneNode(true);
+
+      const markNode = document.createElement('mark');
+      markNode.className = CLASS_NAME;
+      markNode.appendChild(middleClone);
+
+      /* istanbul ignore else */
+      if (middle.parentNode) {
+        middle.parentNode.replaceChild(markNode, middle);
+      }
+
+      return 1;
+    }
+  }
+
+  return 0;
+}
+
+function markTextNodes(node: Node, searchRegex: RegExp): number {
+  if (node.nodeType === 3) {
+    return markNode(node as Text, searchRegex);
+  } else if (node.nodeType === 1 && node.childNodes) {
+    for (let i = 0; i < node.childNodes.length; i++) {
+      const childNode = node.childNodes[i];
+      i += markTextNodes(childNode, searchRegex);
+    }
+  }
+
+  return 0;
+}
+
+function removeHighlight(el: ElementRef): void {
+  const matchedElements = (el.nativeElement as Element).querySelectorAll(
+    `mark.${CLASS_NAME}`
+  );
+
+  if (matchedElements) {
+    for (let i = 0; i < matchedElements.length; i++) {
+      const node = matchedElements[i];
+      const parentNode = node.parentNode;
+
+      if (parentNode && node.firstChild) {
+        parentNode.replaceChild(node.firstChild, node);
+        parentNode.normalize();
+      }
+    }
+  }
+}
+
+function createSearchRegex(searchTerms: string[]): RegExp | undefined {
+  let searchRegex: RegExp | undefined;
+
+  if (searchTerms) {
+    searchTerms = searchTerms.map((searchTerm) =>
+      searchTerm.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&')
```

Comment:
A brief comment describing the purpose of the `RegExp` (or what it's doing) would be helpful for future debugging.

Reply:
Added.